executor/builder.go

```diff
@@ -4149,12 +4146,6 @@ func (builder *dataReaderBuilder) buildTableReaderForIndexJoin(ctx context.Conte
 	}
 	tbl, _ := builder.is.TableByID(tbInfo.ID)
 	pt := tbl.(table.PartitionedTable)
-	pe, err := tbl.(interface {
```

Comment:
Why removing the code here?

Reply:
I think the logic should really be encapsulated with the tables pkg. But I have just gone through the use of this interface check and it is used in several places, so I think it may be better to adjust this patch and then create a separate 'refactor' issue for future versions.

I will change the patch to go with the old style.

Reply:
Got it, thanks!